django/db/models/sql/query.py

```diff
@@ -1178,6 +1178,8 @@ def as_sql(self, compiler, connection):
             and not connection.features.ignores_unnecessary_order_by_in_subqueries
         ):
             self.clear_ordering(force=False)
+            for query in self.combined_queries:
+                query.clear_ordering(force=False)
```

Comment:
Hmmm. What if the ordering in the combined queries is necessary? E.g. if there's also slicing?

Reply:
`force` is set to `False` so ordering is not cleared when the query is sliced, see:
https://github.com/django/django/blob/6f63e0ce8e36fbdb2d0c4dfd77ed3cf02e467f49/django/db/models/sql/query.py#L2217-L2227
I can add an extra test, e.g.
```python
    def test_union_in_with_ordering_and_slice(self):
        qs1 = Number.objects.filter(num__gt=7).order_by("num")[:1]
        qs2 = Number.objects.filter(num__lt=2).order_by("-num")[:1]
        qs = Number.objects.exclude(id__in=qs1.union(qs2).values("id"))
        self.assertNumbersEqual(
            Number.objects.exclude(id__in=qs1.union(qs2).values("id")),
            [0, 2, 3, 4, 5, 6, 7, 9],
            ordered=False,
        )   
```

Reply:
Thanks for the explanation, and yes, I think adding this test is a good idea.

Reply:
Added.